### DISCUSSION TITLE
OpenAI canvas

### DISCUSSION THREAD

**Discussion by S1M0N38** [olimorris/codecompanion.nvim#292] [General]
![Screenshot 2024-10-04 at 10: 11: 04](https://github.com/user-attachments/assets/bc096028-904b-48cb-9b12-76a3b7e8a907)

I've tested a bit of OpenAI canvas, the response to Anthropic artifacts. A feature that I found quite useful is the "Code Review".

"Code Review" goes to your open buffer and identifies lines that can be improved. Then you can incorporate those suggestions individually. "Add logging" is also nice when debugging (it simply adds a print statement in Python).

---

I'm not sure if there is a need to add these features to the core or if they can be implemented as custom actions. I'm just pointing out some workflows that others can find beneficial.

**Comment by olimorris** [comment]
Nice! That looks really slick. Doesn't look like there's an API for this yet but would be really keen to incorporate it into the plugin when it does.

Being able to overlay suggestions on top of the buffer and accept/reject them would be very cool

**Reply by S1M0N38** [reply]
Anthropic does not include artifacts endpoint in their API.

It is reasonable to think that OpenAI will treat canvas in the same way, as a product (i.e. an optimized GPT-4o model with function calling for the "use/edit the canvas" function).

**Reply by olimorris** [reply]
This would be a really interesting tool to create...send the buffer to the LLM to review and critique and overlay the feedback as extmarks
